components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessorTest.java

```diff
@@ -0,0 +1,629 @@
+    private static String BACKCHANNEL_LOGOUT_EVENT = "http://schemas.openid.net/event/backchannel-logout";
+    private static String logoutTokenStatic =
+            "eyJ4NXQiOiJPV0psWmpJME5qSTROR0ZpTVRBNU9UZ3dPR00xTTJJeE5UWmpNekk0TldJeE5EY3dOMkV5TVRNNE5HWmlaVGxoTXpJMFl6a" +
+                    "GpaRFJrWXpoaVl6ZGhPQSIsImtpZCI6Ik9XSmxaakkwTmpJNE5HRmlNVEE1T1Rnd09HTTFNMkl4TlRaak16STROV0l4TkRjd0" +
+                    "4yRXlNVE00TkdaaVpUbGhNekkwWXpoalpEUmtZemhpWXpkaE9BX1JTMjU2IiwiYWxnIjoiUlMyNTYifQ.eyJzdWIiOiJhZG1p" +
+                    "biIsImF1ZCI6IndfSHdwMDVkRlJ3Y1JzX1dGSHY5U053cGZsQWEiLCJpc3MiOiJodHRwczpcL1wvZmVkZXJhdGVkd3NvMi5jb" +
+                    "206OTQ0NFwvb2F1dGgyXC90b2tlbiIsImV4cCI6MTYwOTkxMTk4OCwiaWF0IjoxNjA5OTExODY4LCJqdGkiOiIxNjE1OWUzZS" +
+                    "1jNWZjLTQyZGUtYjkzZi1iMDc4MmFiMzNkNTgiLCJldmVudHMiOnsiaHR0cDpcL1wvc2NoZW1hcy5vcGVuaWRuZXRcL2V2ZW5" +
+                    "0XC9iYWNrY2hhbm5lbC1sb2dvdXQiOnt9fSwic2lkIjoiMTUwNDNmZmMtODc3ZC00MjA1LWFmNDEtOWIxMDdmN2RhMzhjIn0." +
+                    "MG1DbKb4OOMKJ4eIt9FXi8EsppaZgw-PSTmXTD2_ZmGSyApR723J3LZBpsx9StqMzJBJAlXHp9EjFOSeriZv21TIu9zuxHPpK" +
+                    "qEwECJZb21R1Fyeb74O-HEZ0gET3RsuvoIhJk9mXjs7Jcqw0VFfev2bwUSbla5WwwFj3ds7-G31aDew0SDJImiO7MwGdVuQXq" +
+                    "EKgyYA0-FHSbFNRtk3-rN25biW3ivU5AWeo9W3dI6epcNSr4pCCvWBIKI-rk01J8kYyu2ZujecyD0yoz420lbZ2c_dMKFpCDH" +
+                    "DdYjueK4tYE66jpAzvJEyPs37snH-6ok2YaoYjKudyfCdXni7Bg";
+
+    @BeforeTest
+    public void init() throws Exception {
+
+        logoutProcessor = new FederatedIdpInitLogoutProcessor();
+        mockStatic(XMLInputFactory.class);
+        when(XMLInputFactory.newInstance()).thenReturn(new WstxInputFactory());
+        FileBasedConfigurationBuilder.getInstance(TestUtils.getFilePath("application-authentication.xml"));
+        setupIdP();
+    }
+
+    @DataProvider(name = "requestDataHandler")
+    public Object[][] getRequestURI() {
+
+        return new String[][]{
+                // When URI is correct.
+                {"http://localhost:9443/identity/oidc/slo", "true"},
+                {"http://localhost:9443/t/federated.com/identity/oidc/slo", "true"},
+                // When URI is incorrect.
+                {"identity/bclogout", "false"},
+        };
+    }
+
+    @Test(dataProvider = "requestDataHandler")
+    public void testCanHandle(String uri, String expectedCanHandler) {
+
+        when(mockIdentityRequest.getRequestURI()).thenReturn(uri);
+
+        assertEquals(logoutProcessor.canHandle(mockIdentityRequest), Boolean.parseBoolean(expectedCanHandler));
+    }
+
+    @DataProvider(name = "validAudClaimDataHandler")
+    public Object[][] getValidAudClaim() {
+
+        return new String[][]{
+                {"TcEsA1NxUbphVFFFEQIcUxnvtlka", "true"},
+        };
+    }
+
+    @Test(dataProvider = "validAudClaimDataHandler")
+    public void testValidateAud_Valid(String aud, String expectedValidateAud) throws Exception {
```

Comment:
If you are using a single case then getting rid of the data provider will make this much readable